### PR TITLE
CLOUDP-277700: Reduced reconcile period for externally linked resources

### PR DESCRIPTION
**Pull request by helderjs**
### All Submissions:

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes-template.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

**Review by s-urbaniak — APPROVED** [review]

File: pkg/controller/atlasdatabaseuser/atlasdatabaseuser_controller.go [hunk 1]
```diff
@@ -202,6 +202,10 @@ func (r *AtlasDatabaseUserReconciler) ready(ctx *workflow.Context, atlasDatabase
 		EnsureStatusOption(status.AtlasDatabaseUserNameOption(atlasDatabaseUser.Spec.Username)).
 		EnsureStatusOption(status.AtlasDatabaseUserPasswordVersion(passwordVersion))
 
+	if atlasDatabaseUser.Spec.ExternalProjectRef != nil {
+		return workflow.Requeue(workflow.StandaloneResourceRequeuePeriod).ReconcileResult()
+	}
+
 	return workflow.OK().ReconcileResult()
```
Comment (josvazg) [inline pkg/controller/atlasdatabaseuser/atlasdatabaseuser_controller.go:209]: This is the only place we exit OK right?

File: pkg/controller/workflow/result.go [hunk 2]
```diff
@@ -7,8 +7,9 @@ import (
 )
 
 const (
-	DefaultRetry   = time.Second * 10
-	DefaultTimeout = time.Minute * 20
+	DefaultRetry                    = time.Second * 10
+	StandaloneResourceRequeuePeriod = time.Minute * 15
```
Comment (josvazg) [inline pkg/controller/workflow/result.go:11]: We said we would make this configurable down to 5 mins minimum, but not advertised. We can do it on a follow up.

**Review by josvazg — APPROVED** [review]
LGTM
(I expect the CI would eventually pass)

File: pkg/controller/atlasdatabaseuser/atlasdatabaseuser_controller.go (hunk 1 above)
Comment (helderjs) [reply pkg/controller/atlasdatabaseuser/atlasdatabaseuser_controller.go:209]: Yes

File: pkg/controller/workflow/result.go (hunk 2 above)
Comment (josvazg) [reply pkg/controller/workflow/result.go:11]: Thanks for the heads up, I have fixed the TD to reiterate what was said in the scope, this should be configurable before release and 5 minutes is the minimum value allowed.

**Pull request merged**